src/main/java/twilightforest/client/renderer/TileEntityTFChestRenderer.java

```diff
@@ -0,0 +1,155 @@
+package twilightforest.client.renderer;
+
+import net.minecraft.block.Block;
+import net.minecraft.block.BlockChest;
+import net.minecraft.client.model.ModelChest;
+import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.tileentity.TileEntityChest;
+import net.minecraft.util.ResourceLocation;
+
+import org.lwjgl.opengl.GL11;
+import org.lwjgl.opengl.GL12;
+
+import cpw.mods.fml.common.FMLLog;
+import cpw.mods.fml.relauncher.Side;
+import cpw.mods.fml.relauncher.SideOnly;
+import twilightforest.TwilightForestMod;
+import twilightforest.block.BlockTFChest;
+import twilightforest.block.BlockTFChest.WoodType;
+import twilightforest.client.model.ModelTFChest;
+import twilightforest.client.model.ModelTFLargeChest;
+import twilightforest.tileentity.TileEntityTFChest;
+
+@SideOnly(Side.CLIENT)
+public class TileEntityTFChestRenderer extends TileEntitySpecialRenderer {
+
+    public static final String[] names = { "twilight", "canopy", "mangrove", "darkwood", "time", "trans", "mining",
+            "sort" };
+    private static final ResourceLocation[] doubleTextures = new ResourceLocation[names.length];
+    private static final ResourceLocation[] textures = new ResourceLocation[names.length];
+    private ModelChest modelChest = new ModelTFChest();
+    private ModelChest modelLargeChest = new ModelTFLargeChest();
+
+    public TileEntityTFChestRenderer() {
+        for (int i = 0; i < names.length; i++) {
+            doubleTextures[i] = new ResourceLocation(TwilightForestMod.MODEL_DIR + "chest/" + names[i] + "_double.png");
+            textures[i] = new ResourceLocation(TwilightForestMod.MODEL_DIR + "chest/" + names[i] + ".png");
+        }
+    }
+
+    public void renderTileEntityAt(TileEntityChest tileEntity, double x, double y, double z, float f) {
+        int i;
+
+        if (!tileEntity.hasWorldObj()) {
+            i = 0;
+        } else {
+            Block block = tileEntity.getBlockType();
+            i = tileEntity.getBlockMetadata();
+
+            if (block instanceof BlockChest && i == 0) {
+                try {
+                    ((BlockChest) block).func_149954_e(
+                            tileEntity.getWorldObj(),
+                            tileEntity.xCoord,
+                            tileEntity.yCoord,
+                            tileEntity.zCoord);
+                } catch (ClassCastException e) {
+                    FMLLog.severe(
+                            "Attempted to render a chest at %d,  %d, %d that was not a chest",
+                            tileEntity.xCoord,
+                            tileEntity.yCoord,
+                            tileEntity.zCoord);
+                }
+                i = tileEntity.getBlockMetadata();
+            }
+
+            tileEntity.checkForAdjacentChests();
+        }
+
+        BlockTFChest.WoodType type = (((TileEntityTFChest) tileEntity).cachedMaterial == WoodType.NULL)
```

Comment:
This is not thread safe and is throwing class cast exceptions whenever you break a chest with angelica installed. You should have an instanceof or a null check prior to this.
Also if you make a double chest and break the left chest both sides disappear and if you break the right side it continues the render a full double chest.

Reply:
Don't quite get it. What should I check for being null? tileEntity itself?

Reply:
Probably the block you're trying to check but no harm in having null checks for both.

I did however notice something else whilst testing it out
https://github.com/Gordon-Frohman/twilightforest/blob/f9f2e4f9b262db45d2ef65418a87f95f85faca7a/src/main/java/twilightforest/client/renderer/blocks/RenderBlockTFChest.java#L27
I would really not recommend piggybacking off the TESR to render the block in inventory.

Reply:
>I would really not recommend piggybacking off the TESR to render the block in inventory.

I can be wrong, since I'm not near my laptop right now, but if I remember correctly, this code is mostly just a copy+paste of the vanilla chest renderer. Do we really have to change it?

Reply:
Both vanilla and a few other mods do it that way but in most cases it's completely unnecessary for rendering a static item. Anyways you don't have to change it and I'm not gonna be requesting changes for it, I just thought I'd point it out regardless.

Reply:
Got it, thank you for pointing out
Will avoid it in future